### PR TITLE
Install Windows SDK >= 18362 when using VS2019 to fix compilation of UE-4.27.0

### LINKED ISSUE #192: Failing to build UE4.27 image on windows.
Output of the `ue4-docker info` command:

```
ue4-docker version:         0.0.88 (latest available version is 0.0.88)
Operating system:           Windows Server 2019 Standard Evaluation Version 1809 (Build 17763.2114)
Docker daemon version:      20.10.6
NVIDIA Docker supported:    No
Maximum image size:         400GB
Available disk space:       789.61 GiB
Total system memory:        63.92 GiB physical, 73.42 GiB virtual
Number of processors:       12 physical, 24 logical
```

Additional details:

- Are you accessing the network through a proxy server? **No**

Hello, I am having issues building the newest UE 4.27 using the latest ue4-docker (0.0.88).
At first I thought it was due to the custom UE version we use in the company, but I have also tried to build the not customized version from epic and the issue persists. Building the engine using the ```ue4 build-target UE4Editor``` works on both native OS and windows containers. 
Linux image builds just fine.
Previous UE images build just fine.
Older ue4-docker versions (tested 0.0.77 and 0.0.86) have the same issue.
I am able to reproduce the issue on any windows server machine by running:

```powershell
ue4-docker build custom:"4.27" -repo="git@github.com:EpicGames/UnrealEngine.git" -branch="4.27" --visual-studio 2019 --exclude debug --exclude templates --exclude ddc
[ue4-docker build] COMMAND-LINE INVOCATION:
[ue4-docker build] ['ue4-docker', 'custom:4.27', '-repo=git@github.com:EpicGames/UnrealEngine.git', '-branch=4.27', '--visual-studio', '2019', '--exclude', 'debug', '--exclude', 'templates', '--exclude', 'ddc']

[ue4-docker build] UNREAL ENGINE VERSION SETTINGS:
[ue4-docker build] Custom build:  Yes
[ue4-docker build] Custom name:   4.27
[ue4-docker build] Repository:    git@github.com:EpicGames/UnrealEngine.git
[ue4-docker build] Branch/tag:    4.27

[ue4-docker build] ADVANCED CONFIGURATION OPTIONS:
[ue4-docker build] buildgraph_args: " -set:VS2019=true"
[ue4-docker build] excluded_components: {"ddc": true, "debug": true, "templates": true}

[ue4-docker build] WINDOWS CONTAINER SETTINGS
[ue4-docker build] Isolation mode:               process
[ue4-docker build] Base OS image tag:            ltsc2019
[ue4-docker build] Host OS:                      Windows Server 2019 Standard Evaluation Version 1809 (Build 17763.2114)
[ue4-docker build] Memory limit:                 No limit
[ue4-docker build] Detected max image size:      400GB
[ue4-docker build] Visual Studio:                2019
[ue4-docker build] Directory to copy DLLs from:  C:\Windows\System32

[ue4-docker build] GENERAL SETTINGS
[ue4-docker build] Excluding the following Engine components:
[ue4-docker build] - Debug symbols
[ue4-docker build] - Derived Data Cache (DDC)
[ue4-docker build] - Template projects and samples
[ue4-docker build] Not building the ue4-source image, no Git credentials required.

[ue4-docker build] Image "adamrehn/ue4-build-prerequisites:ltsc2019-vs2019" exists and rebuild not requested, skipping build.

[ue4-docker build] Image "adamrehn/ue4-source:4.27-ltsc2019-vs2019" exists and rebuild not requested, skipping build.

[ue4-docker build] Building image "adamrehn/ue4-engine:4.27-ltsc2019-vs2019"...
Sending build context to Docker daemon  3.072kB
Step 1/8 : ARG NAMESPACE
Step 2/8 : ARG TAG
Step 3/8 : ARG PREREQS_TAG
Step 4/8 : FROM ${NAMESPACE}/ue4-source:${TAG}-${PREREQS_TAG}
 ---> 7f4bcdefd16b
Step 5/8 : RUN GenerateProjectFiles.bat && echo. && echo.RUN directive complete. Docker will now commit the filesystem layer to disk. && echo.Note that for large filesystem layers this can take quite some time. && echo.Performing filesystem layer commit... && echo.
 ---> Using cache
 ---> 732cdcb289e5
Step 6/8 : RUN .\Engine\Build\BatchFiles\Build.bat UE4Editor Win64 Development -WaitMutex && echo. && echo.RUN directive complete. Docker will now commit the filesystem layer to disk. && echo.Note that for large filesystem layers this can take quite some time. && echo.Performing filesystem layer commit... && echo.
 ---> Running in 65437974a158
Using 'git status' to determine working set for adaptive non-unity build (C:\UnrealEngine).
Creating makefile for UE4Editor (no existing makefile)
Creating makefile for UnrealHeaderTool (no existing makefile)
Building 48 actions with 24 processes...
  [1/48] Default.rc2
  [2/48] BuildSettings.cpp
  [3/48] UnrealHeaderTool-BuildSettings.lib
     Creating library C:\UnrealEngine\Engine\Intermediate\Build\Win64\UnrealHeaderTool\Development\BuildSettings\UnrealHeaderTool-BuildSettings.lib and object C:\UnrealEngine\Engine\Intermediate\Build\Win64\UnrealHeaderTool\Development\BuildSettings\UnrealHeaderTool-BuildSettings.exp
  [4/48] UnrealHeaderTool-BuildSettings.dll
     Creating library C:\UnrealEngine\Engine\Intermediate\Build\Win64\UnrealHeaderTool\Development\BuildSettings\UnrealHeaderTool-BuildSettings.suppressed.lib and object C:\UnrealEngine\Engine\Intermediate\Build\Win64\UnrealHeaderTool\Development\BuildSettings\UnrealHeaderTool-BuildSettings.suppressed.exp
  [5/48] Module.TraceLog.cpp
  [6/48] UnrealHeaderTool-TraceLog.lib
     Creating library C:\UnrealEngine\Engine\Intermediate\Build\Win64\UnrealHeaderTool\Development\TraceLog\UnrealHeaderTool-TraceLog.lib and object C:\UnrealEngine\Engine\Intermediate\Build\Win64\UnrealHeaderTool\Development\TraceLog\UnrealHeaderTool-TraceLog.exp
  [7/48] UnrealHeaderTool-TraceLog.dll
     Creating library C:\UnrealEngine\Engine\Intermediate\Build\Win64\UnrealHeaderTool\Development\TraceLog\UnrealHeaderTool-TraceLog.suppressed.lib and object C:\UnrealEngine\Engine\Intermediate\Build\Win64\UnrealHeaderTool\Development\TraceLog\UnrealHeaderTool-TraceLog.suppressed.exp
  [8/48] SharedPCH.Core.ShadowErrors.cpp
  [9/48] PCH.Core.cpp
  [10/48] Module.Core.2_of_15.cpp
  [11/48] Module.Projects.cpp
  [12/48] UnrealHeaderTool-Projects.lib
     Creating library C:\UnrealEngine\Engine\Intermediate\Build\Win64\UnrealHeaderTool\Development\Projects\UnrealHeaderTool-Projects.lib and object C:\UnrealEngine\Engine\Intermediate\Build\Win64\UnrealHeaderTool\Development\Projects\UnrealHeaderTool-Projects.exp
  [13/48] PCH.CoreUObject.cpp
  [14/48] Module.Core.11_of_15.cpp
  [15/48] Module.Core.7_of_15.cpp
  [16/48] Module.Json.cpp
  [17/48] Module.Core.4_of_15.cpp
  [18/48] Module.Core.15_of_15.cpp
  [19/48] UnrealHeaderTool-Json.lib
     Creating library C:\UnrealEngine\Engine\Intermediate\Build\Win64\UnrealHeaderTool\Development\Json\UnrealHeaderTool-Json.lib and object C:\UnrealEngine\Engine\Intermediate\Build\Win64\UnrealHeaderTool\Development\Json\UnrealHeaderTool-Json.exp
  [20/48] Module.Core.8_of_15.cpp
  [21/48] Module.Core.3_of_15.cpp
  [22/48] Module.Core.10_of_15.cpp
  [23/48] Module.Core.6_of_15.cpp
  [24/48] Module.CoreUObject.9_of_9.cpp
  [25/48] Module.Core.1_of_15.cpp
  [26/48] Module.Core.13_of_15.cpp
  C:/UnrealEngine/Engine/Source/Runtime/Core/Private/Tests/Math/UnitConversionTests.cpp(143): warning C4305: 'argument': truncation from 'double' to 'float'
  [27/48] Module.Core.9_of_15.cpp
  [28/48] Module.Core.5_of_15.cpp
  [29/48] Module.Core.12_of_15.cpp
  [30/48] Module.Core.14_of_15.cpp
  [31/48] UnrealHeaderTool-Core.lib
     Creating library C:\UnrealEngine\Engine\Intermediate\Build\Win64\UnrealHeaderTool\Development\Core\UnrealHeaderTool-Core.lib and object C:\UnrealEngine\Engine\Intermediate\Build\Win64\UnrealHeaderTool\Development\Core\UnrealHeaderTool-Core.exp
  [32/48] Module.CoreUObject.2_of_9.cpp
  [33/48] UnrealHeaderTool-Json.dll
     Creating library C:\UnrealEngine\Engine\Intermediate\Build\Win64\UnrealHeaderTool\Development\Json\UnrealHeaderTool-Json.suppressed.lib and object C:\UnrealEngine\Engine\Intermediate\Build\Win64\UnrealHeaderTool\Development\Json\UnrealHeaderTool-Json.suppressed.exp
  [34/48] UnrealHeaderTool-Projects.dll
     Creating library C:\UnrealEngine\Engine\Intermediate\Build\Win64\UnrealHeaderTool\Development\Projects\UnrealHeaderTool-Projects.suppressed.lib and object C:\UnrealEngine\Engine\Intermediate\Build\Win64\UnrealHeaderTool\Development\Projects\UnrealHeaderTool-Projects.suppressed.exp
  [35/48] UnrealHeaderTool-Core.dll
     Creating library C:\UnrealEngine\Engine\Intermediate\Build\Win64\UnrealHeaderTool\Development\Core\UnrealHeaderTool-Core.suppressed.lib and object C:\UnrealEngine\Engine\Intermediate\Build\Win64\UnrealHeaderTool\Development\Core\UnrealHeaderTool-Core.suppressed.exp
  [36/48] Module.CoreUObject.6_of_9.cpp
  [37/48] Module.CoreUObject.7_of_9.cpp
  [38/48] Module.CoreUObject.4_of_9.cpp
  [39/48] Module.CoreUObject.3_of_9.cpp
  [40/48] Module.CoreUObject.1_of_9.cpp
  [41/48] Module.CoreUObject.5_of_9.cpp
  [42/48] Module.UnrealHeaderTool.cpp
  C:/UnrealEngine/Engine/Source/Programs/UnrealHeaderTool/Private/CodeGenerator.cpp(2465): warning C4305: 'argument': truncation from 'double' to 'float'
  C:/UnrealEngine/Engine/Source/Programs/UnrealHeaderTool/Private/CodeGenerator.cpp(3395): warning C4305: 'argument': truncation from 'double' to 'float'
  [43/48] UnrealHeaderTool.lib
     Creating library C:\UnrealEngine\Engine\Intermediate\Build\Win64\UnrealHeaderTool\Development\UnrealHeaderTool.lib and object C:\UnrealEngine\Engine\Intermediate\Build\Win64\UnrealHeaderTool\Development\UnrealHeaderTool.exp
  [44/48] Module.CoreUObject.8_of_9.cpp
  [45/48] UnrealHeaderTool-CoreUObject.lib
     Creating library C:\UnrealEngine\Engine\Intermediate\Build\Win64\UnrealHeaderTool\Development\CoreUObject\UnrealHeaderTool-CoreUObject.lib and object C:\UnrealEngine\Engine\Intermediate\Build\Win64\UnrealHeaderTool\Development\CoreUObject\UnrealHeaderTool-CoreUObject.exp
  [46/48] UnrealHeaderTool-CoreUObject.dll
     Creating library C:\UnrealEngine\Engine\Intermediate\Build\Win64\UnrealHeaderTool\Development\CoreUObject\UnrealHeaderTool-CoreUObject.suppressed.lib and object C:\UnrealEngine\Engine\Intermediate\Build\Win64\UnrealHeaderTool\Development\CoreUObject\UnrealHeaderTool-CoreUObject.suppressed.exp
  [47/48] UnrealHeaderTool.exe
  [48/48] UnrealHeaderTool.target
Total time in Parallel executor: 29.33 seconds
Parsing headers for UE4Editor
  Running UnrealHeaderTool UE4Editor "C:\UnrealEngine\Engine\Intermediate\Build\Win64\UE4Editor\Development\UE4Editor.uhtmanifest" -LogCmds="loginit warning, logexit warning, logdatabase error" -Unattended -WarningsAsErrors -abslog="C:\UnrealEngine\Engine\Programs\UnrealBuildTool\Log_UHT.txt"
LogInit: Display: Loading text-based GConfig....
Reflection code generated for UE4Editor in 12.7497078 seconds
Building UE4Editor...
Using Visual Studio 2019 14.29.30133 toolchain (C:\BuildTools\VC\Tools\MSVC\14.29.30133) and Windows 10.0.17763.0 SDK (C:\Program Files (x86)\Windows Kits\10).
Building 4561 actions with 24 processes...
  [1/4561] PBDSpringConstraints.ispc
  [2/4561] GeometryCollectionSceneProxy.ispc
  [3/4561] GPUSkinVertexFactory.ispc
  [4/4561] AABB.ispc
  [5/4561] PBDAxialSpringConstraints.ispc
  [6/4561] PBDLongRangeConstraints.ispc
  [7/4561] BonePose.ispc
  [8/4561] SkeletalMeshComponent.ispc
  [9/4561] ChaosClothingSimulation.ispc
  [10/4561] VelocityField.ispc
  [11/4561] NiagaraDataInterfaceVectorField.ispc
  [12/4561] TriangleMesh.ispc
  [13/4561] AnimationRuntime.ispc
  [14/4561] KAggregateGeom.ispc
  [15/4561] GeometryCollectionComponent.ispc
  [16/4561] PBDSphericalConstraint.ispc
  [17/4561] PBDMinEvolution.ispc
  [18/4561] GPUSkinVertexFactory.ispc
  [19/4561] Default.rc2
  [20/4561] GeometryCollectionSceneProxy.ispc
  [21/4561] AABB.ispc
  [22/4561] PerParticleDampVelocity.ispc
  [23/4561] PBDSpringConstraints.ispc
  [24/4561] PBDLongRangeConstraints.ispc
  [25/4561] BonePose.ispc
  [26/4561] PBDAxialSpringConstraints.ispc
  [27/4561] SkeletalMeshComponent.ispc
  [28/4561] AnimEncoding_ConstantKeyLerp.ispc
  [29/4561] VelocityField.ispc
  [30/4561] TriangleMesh.ispc
  [31/4561] ChaosClothingSimulation.ispc
  [32/4561] NiagaraDataInterfaceVectorField.ispc
  [33/4561] PCH.ImageWrapper.cpp
  [34/4561] SharedPCH.CoreUObject.ShadowErrors.cpp
  [35/4561] SharedPCH.Core.ShadowErrors.cpp
  [36/4561] AnimationRuntime.ispc
  [37/4561] KAggregateGeom.ispc
  [38/4561] PCH.Core.cpp
  [39/4561] AnimEncoding_VariableKeyLerp.ispc
  [40/4561] PBDSphericalConstraint.ispc
  [41/4561] Module.TraceLog.cpp
  [42/4561] GeometryCollectionComponent.ispc
  [43/4561] PBDMinEvolution.ispc
  [44/4561] PCH.CoreUObject.cpp
  [45/4561] PBDJointSolverGaussSeidel.ispc
  [46/4561] Module.ImageWrapper.cpp
  [47/4561] PerParticlePBDCollisionConstraint.ispc
  [48/4561] PerParticleDampVelocity.ispc
  [49/4561] Module.SlateCore.gen.cpp
  [50/4561] AnimEncoding_PerTrackCompression.ispc
  [51/4561] Module.Core.7_of_15.cpp
  [52/4561] Module.Core.10_of_15.cpp
  [53/4561] AnimEncoding_ConstantKeyLerp.ispc
  [54/4561] SharedPCH.Slate.ShadowErrors.cpp
  [55/4561] Module.SlateCore.2_of_3.cpp
.
.
.
  [4543/4561] UE4Editor-DisplayClusterConfigurator.dll
     Creating library C:\UnrealEngine\Engine\Plugins\Runtime\nDisplay\Intermediate\Build\Win64\UE4Editor\Development\DisplayClusterConfigurator\UE4Editor-DisplayClusterConfigurator.suppressed.lib and object C:\UnrealEngine\Engine\Plugins\Runtime\nDisplay\Intermediate\Build\Win64\UE4Editor\Development\DisplayClusterConfigurator\UE4Editor-DisplayClusterConfigurator.suppressed.exp
  [4544/4561] UE4Editor-DatasmithPLMXMLTranslator.dll
     Creating library C:\UnrealEngine\Engine\Plugins\Enterprise\DatasmithCADImporter\Intermediate\Build\Win64\UE4Editor\Development\DatasmithPLMXMLTranslator\UE4Editor-DatasmithPLMXMLTranslator.suppressed.lib and object C:\UnrealEngine\Engine\Plugins\Enterprise\DatasmithCADImporter\Intermediate\Build\Win64\UE4Editor\Development\DatasmithPLMXMLTranslator\UE4Editor-DatasmithPLMXMLTranslator.suppressed.exp
  [4545/4561] UE4Editor-MeshModelingToolsEditorOnly.dll
     Creating library C:\UnrealEngine\Engine\Plugins\Experimental\MeshModelingToolset\Intermediate\Build\Win64\UE4Editor\Development\MeshModelingToolsEditorOnly\UE4Editor-MeshModelingToolsEditorOnly.suppressed.lib and object C:\UnrealEngine\Engine\Plugins\Experimental\MeshModelingToolset\Intermediate\Build\Win64\UE4Editor\Development\MeshModelingToolsEditorOnly\UE4Editor-MeshModelingToolsEditorOnly.suppressed.exp
  [4546/4561] UE4Editor-MagicLeapAR.dll
     Creating library C:\UnrealEngine\Engine\Plugins\Lumin\MagicLeap\Intermediate\Build\Win64\UE4Editor\Development\MagicLeapAR\UE4Editor-MagicLeapAR.suppressed.lib and object C:\UnrealEngine\Engine\Plugins\Lumin\MagicLeap\Intermediate\Build\Win64\UE4Editor\Development\MagicLeapAR\UE4Editor-MagicLeapAR.suppressed.exp
  [4547/4561] UE4Editor-GameplayAbilitiesEditor.dll
     Creating library C:\UnrealEngine\Engine\Plugins\Runtime\GameplayAbilities\Intermediate\Build\Win64\UE4Editor\Development\GameplayAbilitiesEditor\UE4Editor-GameplayAbilitiesEditor.suppressed.lib and object C:\UnrealEngine\Engine\Plugins\Runtime\GameplayAbilities\Intermediate\Build\Win64\UE4Editor\Development\GameplayAbilitiesEditor\UE4Editor-GameplayAbilitiesEditor.suppressed.exp
  [4548/4561] UE4Editor.exe
     Creating library C:\UnrealEngine\Engine\Intermediate\Build\Win64\UE4Editor\Development\UE4Editor.suppressed.lib and object C:\UnrealEngine\Engine\Intermediate\Build\Win64\UE4Editor\Development\UE4Editor.suppressed.exp
  [4549/4561] UE4Editor-VirtualCamera.dll
     Creating library C:\UnrealEngine\Engine\Plugins\Experimental\VirtualCamera\Intermediate\Build\Win64\UE4Editor\Development\VirtualCamera\UE4Editor-VirtualCamera.suppressed.lib and object C:\UnrealEngine\Engine\Plugins\Experimental\VirtualCamera\Intermediate\Build\Win64\UE4Editor\Development\VirtualCamera\UE4Editor-VirtualCamera.suppressed.exp
  [4550/4561] UE4Editor-PixelStreaming.dll
     Creating library C:\UnrealEngine\Engine\Plugins\Media\PixelStreaming\Intermediate\Build\Win64\UE4Editor\Development\PixelStreaming\UE4Editor-PixelStreaming.suppressed.lib and object C:\UnrealEngine\Engine\Plugins\Media\PixelStreaming\Intermediate\Build\Win64\UE4Editor\Development\PixelStreaming\UE4Editor-PixelStreaming.suppressed.exp
  [4551/4561] UE4Editor-DatasmithImporter.dll
     Creating library C:\UnrealEngine\Engine\Plugins\Enterprise\DatasmithImporter\Intermediate\Build\Win64\UE4Editor\Development\DatasmithImporter\UE4Editor-DatasmithImporter.suppressed.lib and object C:\UnrealEngine\Engine\Plugins\Enterprise\DatasmithImporter\Intermediate\Build\Win64\UE4Editor\Development\DatasmithImporter\UE4Editor-DatasmithImporter.suppressed.exp
  [4552/4561] UE4Editor-USDExporter.dll
     Creating library C:\UnrealEngine\Engine\Plugins\Importers\USDImporter\Intermediate\Build\Win64\UE4Editor\Development\USDExporter\UE4Editor-USDExporter.suppressed.lib and object C:\UnrealEngine\Engine\Plugins\Importers\USDImporter\Intermediate\Build\Win64\UE4Editor\Development\USDExporter\UE4Editor-USDExporter.suppressed.exp
  [4553/4561] UE4Editor-CameraCalibrationEditor.dll
     Creating library C:\UnrealEngine\Engine\Plugins\VirtualProduction\CameraCalibration\Intermediate\Build\Win64\UE4Editor\Development\CameraCalibrationEditor\UE4Editor-CameraCalibrationEditor.suppressed.lib and object C:\UnrealEngine\Engine\Plugins\VirtualProduction\CameraCalibration\Intermediate\Build\Win64\UE4Editor\Development\CameraCalibrationEditor\UE4Editor-CameraCalibrationEditor.suppressed.exp
  [4554/4561] UE4Editor-ControlRigEditor.dll
     Creating library C:\UnrealEngine\Engine\Plugins\Experimental\ControlRig\Intermediate\Build\Win64\UE4Editor\Development\ControlRigEditor\UE4Editor-ControlRigEditor.suppressed.lib and object C:\UnrealEngine\Engine\Plugins\Experimental\ControlRig\Intermediate\Build\Win64\UE4Editor\Development\ControlRigEditor\UE4Editor-ControlRigEditor.suppressed.exp
  [4555/4561] UE4Editor-DetailCustomizations.dll
     Creating library C:\UnrealEngine\Engine\Intermediate\Build\Win64\UE4Editor\Development\DetailCustomizations\UE4Editor-DetailCustomizations.suppressed.lib and object C:\UnrealEngine\Engine\Intermediate\Build\Win64\UE4Editor\Development\DetailCustomizations\UE4Editor-DetailCustomizations.suppressed.exp
  [4556/4561] UE4Editor-NiagaraEditor.dll
     Creating library C:\UnrealEngine\Engine\Plugins\FX\Niagara\Intermediate\Build\Win64\UE4Editor\Development\NiagaraEditor\UE4Editor-NiagaraEditor.suppressed.lib and object C:\UnrealEngine\Engine\Plugins\FX\Niagara\Intermediate\Build\Win64\UE4Editor\Development\NiagaraEditor\UE4Editor-NiagaraEditor.suppressed.exp
  [4557/4561] UE4Editor-UnrealEd.dll
     Creating library C:\UnrealEngine\Engine\Intermediate\Build\Win64\UE4Editor\Development\UnrealEd\UE4Editor-UnrealEd.suppressed.lib and object C:\UnrealEngine\Engine\Intermediate\Build\Win64\UE4Editor\Development\UnrealEd\UE4Editor-UnrealEd.suppressed.exp
The command 'cmd /S /C .\Engine\Build\BatchFiles\Build.bat UE4Editor Win64 Development -WaitMutex && echo. && echo.RUN directive complete. Docker will now commit the filesystem layer to disk. && echo.Note that for large filesystem layers this can take quite some time. && echo.Performing filesystem layer commit... && echo.' returned a non-zero code: 6
[ue4-docker build] Error: failed to build image "adamrehn/ue4-engine:4.27-ltsc2019-vs2019".
```
Thanks in advance

### PR DESCRIPTION
**Pull request by slonopotamus**
resolves #192

----

This is an alternative to #193. See discussion in #192.

P.S. Testing is not finished yet.

### CODE REVIEW

File: ue4docker/dockerfiles/ue4-build-prerequisites/windows/install-prerequisites.bat [hunk 1]
```diff
@@ -33,6 +33,29 @@ powershell -Command "Copy-Item -Path \"*\x64\vulkan-1.dll\" -Destination C:\Wind
...
+@rem See https://github.com/EpicGames/UnrealEngine/commit/ecc4872c3269e75a24adc40734cc8bcc9bbed1ca
+@rem See https://udn.unrealengine.com/s/question/0D54z000079HcjJCAS/d3d12h427-error-c4668-winapipartitiongames-is-not-defined-as-a-preprocessor-macro-replacing-with-0-for-ifelif
+@rem
+@rem Keywords for Google:
+@rem error C4668: 'NDIS_MINIPORT_MAJOR_VERSION' is not defined as a preprocessor macro, replacing with '0' for '#if/#elif
+@rem d3d12.h(427): error C4668: 'WINAPI_PARTITION_GAMES' is not defined as a preprocessor macro, replacing with '0' for '#if/#elif'
+if "%VISUAL_STUDIO_BUILD_NUMBER%" == "15" (
+    set WINDOWS_SDK_VERSION=17763
+) else (
+    set WINDOWS_SDK_VERSION=20348
```
[the remark this answers is not in the document]
Comment (TBBle) [reply ue4docker/dockerfiles/ue4-build-prerequisites/windows/install-prerequisites.bat:56]: Looking at the [release notes](https://developer.microsoft.com/en-US/windows/downloads/windows-10-sdk/#relnote), 20348 gets some C++ improvements and features, which is nice, but seems to have some Clang/ARM64 issues. I'm not sure if that's a target UE4 currently supports though.

Looking at the specific issues in more detail, if UE4 uses DirectXMath, I expect they use a bundled version anyway, and I can't see that they'd use a Windows 11 target anywhere.

So I'd go with 20348 barring any kind of compile issue showing up.

File: ue4docker/dockerfiles/ue4-build-prerequisites/windows/install-prerequisites.bat [hunk 2]
```diff
@@ -33,6 +33,29 @@ powershell -Command "Copy-Item -Path \"*\x64\vulkan-1.dll\" -Destination C:\Wind
...
+@rem
+@rem See https://github.com/adamrehn/ue4-docker/issues/192
+@rem See https://forums.unrealengine.com/t/ndis_miniport_major_version-is-not-defined-error/135058
+@rem See https://github.com/EpicGames/UnrealEngine/blame/4.23.0-release/Engine/Source/Programs/UnrealBuildTool/Platform/Windows/UEBuildWindows.cs#L1822-L1823
+@rem See https://github.com/EpicGames/UnrealEngine/commit/ecc4872c3269e75a24adc40734cc8bcc9bbed1ca
+@rem See https://udn.unrealengine.com/s/question/0D54z000079HcjJCAS/d3d12h427-error-c4668-winapipartitiongames-is-not-defined-as-a-preprocessor-macro-replacing-with-0-for-ifelif
+@rem
+@rem Keywords for Google:
+@rem error C4668: 'NDIS_MINIPORT_MAJOR_VERSION' is not defined as a preprocessor macro, replacing with '0' for '#if/#elif
+@rem d3d12.h(427): error C4668: 'WINAPI_PARTITION_GAMES' is not defined as a preprocessor macro, replacing with '0' for '#if/#elif'
```
Comment (TBBle) [inline ue4docker/dockerfiles/ue4-build-prerequisites/windows/install-prerequisites.bat:52]: Ohhh, source-code SEO? Is this Web 7.0?

**Review by TBBle — APPROVED** [review]
At some point, users may want the SDK version to be configurable. But not worth doing yet.

I like this approach, the only risk I see is if a project is using VS2019-incompatible code in their own source in a UE 4.27 project. That'd require a pretty complicated setup to work _locally_, and unless we find a real person doing this, it's not worth trying to complicate things here.

**Comment by TBBle** [comment]
Should we also implement `MaxSupportedUnreal` to go with [`MinSupportedUnreal`](https://github.com/adamrehn/ue4-docker/blob/b82431c582e10231d4885224011a4f41af37698d/ue4docker/infrastructure/BuildConfiguration.py#L42-L49)?

Or is that too restrictive given both 4.27.1, and the possibility to locally patch 4.27.0 to work with the old SDK and hence support the VS2017 pre-reqs.

I've just realised that (from brief code-inspection) we don't block UE5 with VS2017 either.

**Comment by slonopotamus** [comment]
> Should we also implement MaxSupportedUnreal to go with MinSupportedUnreal?

We should at one point, but not yet. UE5 can only currently be built as a "custom version", so we effectively treat its version string as an opaque identifier and cannot do conditions with it.

**Comment by alexgeek** [comment]
I tried this branch as I need to build 4.27.0 but it still fails for me:

```
UE4Editor-Cmd.exe
     Creating library C:\UnrealEngine\Engine\Intermediate\Build\Win64\UE4Editor\Development\UE4Editor-Cmd.suppressed.lib and object C:\UnrealEngine\Engine\Intermediate\Build\Win64\UE4Editor\Development\UE4Editor-Cmd.suppressed.exp
  UE4Editor.exe
     Creating library C:\UnrealEngine\Engine\Intermediate\Build\Win64\UE4Editor\Development\UE4Editor.suppressed.lib and object C:\UnrealEngine\Engine\Intermediate\Build\Win64\UE4Editor\Development\UE4Editor.suppressed.exp
BUILD FAILED:  failed, retries not enabled:
AutomationTool exiting with ExitCode=1 (Error_Unknown)
BUILD FAILED
The command 'cmd /S /C .\Engine\Build\BatchFiles\RunUAT.bat BuildGraph -target="Make Installed Build Win64" -script=Engine/Build/InstalledEngineBuild.xml -set:HostPlatformOnly=true -set:WithDDC=false  &&     (if exist C:\UnrealEngine\LocalBuilds\InstalledDDC rmdir /s /q C:\UnrealEngine\LocalBuilds\InstalledDDC) && echo. && echo.RUN directive complete. Docker will now commit the filesystem layer to disk. && echo.Note that for large filesystem layers this can take quite some time. && echo.Performing filesystem layer commit... && echo.' returned a non-zero code: 1
```

With command:
```
ue4-docker build 4.27.0 --no-engine --exclude ddc --exclude templates -username alexgeek -password [personal access token]
```

**Comment by slonopotamus** [comment]
I've successfully build 4.27.0 and 5.0EA2 using these changes, seems good enough to merge.

**Pull request merged**